### PR TITLE
cleanup(virtctl,params): Add unit tests and apply cleanups

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Add unit tests for pkg/virtctl/create/params that capture the current
behavior of the module.

Then apply cleanups to the module:
- Do not panic in Map, since it is able to return an error. Supported now always panics with a wrapped error.
- Return the correct count of slashes in errors returned by params.SplitPrefixedName. It was off by one previously.
- Remove named return in params.SplitPrefixedName.
- Add the missing support for string slices in params.Map.

Also apply these minor cleanups to the module:
- Rename variables to make the code easier to read
- Name variables consistently
- Inline assignments if possible
- Use pointer.P
- Extend comment of params.Map to clarify its behavior

Before this PR:

pkg/virtctl/create/params has no unit tests and needs cleanups

After this PR:

pkg/virtctl/create/params is now tested by unit tests and cleanups were applied

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virtctl/create/params/params_test.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+	Context("FlagErr", func() {
+		const (
+			format = "My wrapped test error: %s"
+			value  = "testvalue"
+		)
+
+		It("should have correct message", func() {
+			err := params.FlagErr(flagName, format, value)
+			Expect(err).To(MatchError(ContainSubstring(fmt.Sprintf("failed to parse \"--%s\" flag: ", flagName))))
```
Comment (lyarwood) [inline pkg/virtctl/create/params/params_test.go:72]: nit - Would you mind sharing these strings as public constants from the params package?

**Review by lyarwood — COMMENTED** [review]
Looks good overall, I've got one nit about providing the strings you're asserting in the tests as public constants from the package. IMHO this just makes things more maintainable moving forward but it isn't a massive issue.

**Review by lyarwood — APPROVED** [review]
/lgtm

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**